scalableminds/webknossos
# Tensorflow segmentation

Open Issues:
- [x] Delete old z-slice predictions from the predictions map to free up space
- [x] Multiple Magic Wand "instances" can be spawned
- [ ] Magic Wand only works in "positive" z-direction (maybe fine for now)
- [x] Should the Magic Wand be disabled by default and only be enabled via dev console or feature flag?
- [x] Update the tensorflow model to the fully trained new version

### URL of deployed dev instance (used for testing):
- https://tensorflowsegmentation.webknossos.xyz

### Steps to test:
- Open a volume tracing.
- Use `Ctrl + Click` on a segment to trigger the Magic Wand. A toast should open and inform you of the process.
- Use the brush to correct the prediction and move in positive z-direction through the dataset.
- The magic wand mode should always predict the next 3 slices, then wait until the user moved "forward" and then continue predicting.
- Close the modal to stop the magic wand mode.
- The magic wand should also work on the XZ and YZ viewports, but the current model doesn't perform very well on these (will be improved with the new model).
- Press `Alt` in any mode to move the viewport in a photoshop-like manner.

### Issues:
- fixes #3960 

------
~~- [ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
~~- [ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- [x] add new routes to wk-connect or [document the need](https://github.com/scalableminds/webknossos-connect/issues/22)
